get_openstack_info_open_port/get_openstack_info_open_port/get_openstack_info_open_port.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python3
+# -*- encoding: utf-8 -*-
+#
+# This script is licensed under GNU GPL version 2.0 or above
+# (c) 2021 Antonio J. Delgado
+# __description__
+
+import sys
+import os
+import logging
+import click
+import click_config_file
+from logging.handlers import SysLogHandler
+from keystoneauth1 import session
+from keystoneauth1.identity import v3
+from keystoneclient.v3 import client as keystoneclient_v3
+from novaclient import client as nova
+from neutronclient.neutron import client as neutron
+import ipaddress
+import json
+import socket
+
+class CustomFormatter(logging.Formatter):
+    """Logging colored formatter, adapted from https://stackoverflow.com/a/56944256/3638629"""
+
+    grey = '\x1b[38;21m'
+    blue = '\x1b[38;5;39m'
+    yellow = '\x1b[38;5;226m'
+    red = '\x1b[38;5;196m'
+    bold_red = '\x1b[31;1m'
+    reset = '\x1b[0m'
+
+    def __init__(self, fmt):
+        super().__init__()
+        self.fmt = fmt
+        self.FORMATS = {
+            logging.DEBUG: self.grey + self.fmt + self.reset,
+            logging.INFO: self.blue + self.fmt + self.reset,
+            logging.WARNING: self.yellow + self.fmt + self.reset,
+            logging.ERROR: self.red + self.fmt + self.reset,
+            logging.CRITICAL: self.bold_red + self.fmt + self.reset
+        }
+
+    def format(self, record):
+        log_fmt = self.FORMATS.get(record.levelno)
+        formatter = logging.Formatter(log_fmt)
+        return formatter.format(record)
+
+def jprint(object):
+    print(json.dumps(object, indent=2))
+    
+class get_openstack_info_open_port:
+
+    def __init__(self, debug_level, log_file, ip, port, method):
+        ''' Initial function called when object is created '''
+        self.config = dict()
+        self.config['debug_level'] = debug_level
+        if log_file is None:
+            log_file = os.path.join(os.environ.get('HOME', os.environ.get('USERPROFILE', os.getcwd())), 'log', 'get_openstack_info_open_port.log')
+        self.config['log_file'] = log_file
+        self.ip = ip
+        self.port = port
+        self.rule_found = False
+        self._init_log()
+        self._os_auth()
+        self.all_floating_ips = self.neutron.list_floatingips()['floatingips']
+        if self._check_known_ip():
+            self._log.debug('IP found in your environment.')
+            self._test_open_port()
+            self._check_openstack()
```

Comment:
More verbose out put would be better.
Typically
project name / id
server name / id
This would give us the info to then be able to contact the customer / make things a bit easier.

Reply:
Added with some workaround with the project id. I'm getting it from the IP addresses, that should only show a project (but it's a dictionary with the project name as key so, it might be possible to have IP's from different project?). There is no python client with Placement to query resources placement, and I won't add (for now) direct calls to API that might be in some module hidden somewhere